src/IK/include/BiomechanicalAnalysis/IK/InverseKinematics.h

```diff
@@ -113,13 +129,45 @@ class HumanIK
     bool setInitialJointPositions(const Eigen::Ref<const Eigen::VectorXd> qInitial);
 
     /**
-     * set the orientation and angular velocity for the link 1 task
-     * @param link1Orientation link 1 orientation
-     * @param link1AngularVelocity link 1 angular velocity
-     * @return true if the link 1 orientation and angular velocity are set correctly
+     * set the orientation and the angular velocity of a given node
+     * @param node node number
+     * @param I_R_IMU orientation of the IMU
+     * @param I_omega_IMU angular velocity of the IMU
     */
-    bool setLink1OrientationAndAngVel(const manif::SO3d &link1Orientation,
-                                      const manif::SO3Tangentd &link1AngularVelocity);
+    bool setNodeSetPoint(int node,const iDynTree::Rotation &I_R_IMU,
+                                           const iDynTree::AngVelocity &I_omega_IMU);
+
+    // set the set point for the orientation tasks
+    bool setPelvisSetPoint(const manif::SO3d &pelvisOrientation,
+                                      const manif::SO3Tangentd &pelvisAngularVelocity);
+
+    bool setT8SetPoint(const manif::SO3d &T8Orientation,
+                                      const manif::SO3Tangentd &T8AngularVelocity);
+
+    bool setRightUpperArmSetPoint(const manif::SO3d &RightUpperArmOrientation,
+                                        const manif::SO3Tangentd &RightUpperArmAngularVelocity);
+
+    bool setRightForeArmSetPoint(const manif::SO3d &RightForeArmOrientation,
+                                        const manif::SO3Tangentd &RightForeArmAngularVelocity);
+
+    bool setLeftUpperArmSetPoint(const manif::SO3d &LeftUpperArmOrientation,
+                                        const manif::SO3Tangentd &LeftUpperArmAngularVelocity);
+
+    bool setLeftForeArmSetPoint(const manif::SO3d &LeftForeArmOrientation,
+                                        const manif::SO3Tangentd &LeftForeArmAngularVelocity);
+
+    bool setRightUpperLegSetPoint(const manif::SO3d &RightUpperLegOrientation,
+                                        const manif::SO3Tangentd &RightUpperLegAngularVelocity);
+
+    bool setRightLowerLegSetPoint(const manif::SO3d &RightLowerLegOrientation,
+                                        const manif::SO3Tangentd &RightLowerLegAngularVelocity);
+
+    bool setLeftUpperLegSetPoint(const manif::SO3d &LeftUpperLegOrientation,
+                                        const manif::SO3Tangentd &LeftUpperLegAngularVelocity);
+
+    bool setLeftLowerLegSetPoint(const manif::SO3d &LeftLowerLegOrientation,
```

Comment:
Similar to @GiulioRomualdi comment before, perhaps a `setPoeSetPoint(const std::string key, orientation, velocity)` generic method may be more easy to use for users and also more generic?

Reply:
Yes, I created this function:
https://github.com/ami-iit/biomechanical-analysis-framework/blob/c1d12c4d8edf191e82688d9bca959b97374974cc/src/IK/include/BiomechanicalAnalysis/IK/InverseKinematics.h#L122-L129
to set the setpoint of a node specifying the number of the node.